src/ngraph/runtime/gpu/gpu_kernel_emitters.cpp

```diff
@@ -45,6 +45,129 @@ void runtime::gpu::kernel::emit_memcpyDtD(codegen::CodeWriter& writer,
+    }
+    else
+    {
+        writer << "const int " << name << "_padding[] = {" << join(padding) << "};\n";
+        writer << "const int " << name << "_window_movement_strides[] = {"
+               << join(window_movement_strides) << "};\n";
+        writer << "const int " << name << "_window_dilation_strides[] = {"
+               << join(window_dilation_strides) << "};\n";
+
+        writer << "cudnnSetConvolutionNdDescriptor(" << name << ", " << padding.size() << ", "
+               << name << "_padding, " << name << "_window_movement_strides, " << name
+               << "_window_dilation_strides, " << mode << ", " << data_type << ");\n";
+    }
+}
+
+void runtime::gpu::kernel::emit_cudnnFilterDescriptor(codegen::CodeWriter& writer,
+                                                      const std::string& name,
+                                                      const std::string& format,
+                                                      const std::string& data_type,
+                                                      const Shape& shape)
+{
+    std::vector<size_t> dimensions;
+    for (size_t i = shape.size(); i < 4; i++)
+    {
+        dimensions.push_back(1);
+    }
+    for (size_t i = 0; i < shape.size(); i++)
+    {
+        dimensions.push_back(shape[i]);
+    }
+
+    writer << "cudnnFilterDescriptor_t " << name << ";\n";
+    writer << "cudnnCreateFilterDescriptor(&" << name << ");\n";
+
+    if (dimensions.size() <= 4)
+    {
+        writer << "cudnnSetFilter4dDescriptor(" << name << ",\n";
+        writer << "                 /*dataType=*/" << data_type << ",\n";
+        writer << "                 /*format=*/" << format << ",\n";
+        writer << "                 /*dimension_size*/" << dimensions[0] << ",\n";
+        writer << "                 /*dimension_size*/" << dimensions[1] << ",\n";
+        writer << "                 /*dimension_size*/" << dimensions[2] << ",\n";
+        writer << "                 /*dimension_size*/" << dimensions[3] << ");\n";
+    }
+    else
+    {
+        writer << "const int " << name << "_axes[] = {" << join(dimensions) << "};\n";
+        writer << "cudnnSetFilterNdDescriptor(" << name << ",\n";
+        writer << "                 /*dataType=*/" << data_type << ",\n";
+        writer << "                 /*format=*/" << format << ",\n";
+        writer << "                 /*num_dimensions=*/" << dimensions.size() << ",\n";
+        writer << "                 /*dimensions*/" << name << "_axes);\n";
+    }
+}
+
+void runtime::gpu::kernel::emit_cudnnTensorDescriptor(codegen::CodeWriter& writer,
+                                                      const std::string& name,
+                                                      const std::string& format,
+                                                      const std::string& data_type,
+                                                      const Shape& shape)
+{
+    std::vector<size_t> dimensions;
```

Comment:
nitpick: this just pads `dimensions` to at least 4 (if less) with 1s? I wonder if we could add a short comment?

Reply:
Or just initialize dimensions with {1, 1, 1, 1}